### ISSUE TITLE
Add support for custom ID to be used by Java Mail on MimeMessage

### ISSUE THREAD

**Issue by bbottema** [bbottema/simple-java-mail#79]
See #77 for requirement details.

Need to solve this properly though. Email is the Simple Java Mail abstraction to make the underlying API simple. Exposing MimeMessage in Email would be doing the opposite.

Instead Email will have an ID field, which will be used if provided.

Note though that the ID needs to be conform https://tools.ietf.org/html/rfc5322#section-3.6.4

**Comment by bbottema** [comment]
Released in 4.2.3

**Comment by bbottema** [comment]
Current 9.x usage (verified against 9.3.2):

```java
Email email = EmailBuilder.startingBlank()
    .fixingMessageId("<123@456>")
    .buildEmail();
```

The mutable `Email#setId` and old `.id(...)` builder call were replaced by the builder API. The supplied value must be a valid RFC 5322 message-id. See the [5.0 builder migration](https://www.simplejavamail.org/migration-notes-5.0.0.html#section-what-changed) and [current custom-ID documentation](https://www.simplejavamail.org/features.html#section-custom-id).
